### DISCUSSION TITLE
Setting $releasever, $arch, and $basearch with Python bindings

### DISCUSSION THREAD

**Discussion by gotmax23** [rpm-software-management/dnf5#232] [Q&A] [answered]
Hi! I couldn't figure out how to get or set/override these values with the libdnf5 Python bindings. Can you point me in the right direction?

**Comment by gotmax23** [comment]
Hmm, I guess you need to run `base.load_config_from_file()` and `base.setup()` before `base.get_vars().get_value(...)`.

```
$ ipython
In [1]: import libdnf5

In [2]: base = libdnf5.base.Base()

In [3]: base.load_config_from_file()

In [4]: base.get_vars().get_value("arch")
terminate called after throwing an instance of 'std::out_of_range'
  what():  map::at
zsh: IOT instruction (core dumped)  /usr/bin/ipython
```

```
$ ipython

In [1]: import libdnf5

In [2]: base = libdnf5.base.Base()

In [3]: base.load_config_from_file()

In [4]: base.setup()

In [5]: base.get_vars().get_value("arch")
Out[5]: 'x86_64'

In [6]: base.get_vars().get_value("releasever")
Out[6]: '37'

```

This should be documented and probably should raise an Exception instead of crashing the entire python interpreter when someone does the wrong thing.

**Reply by inknos** [reply]
Thank you for your report, we are ultimately willing to look into that. I would move the discussion into an issue, and we can continue from there.

**Comment by gotmax23** [comment, marked as answer]
Closing in favor of https://github.com/rpm-software-management/dnf5/issues/241. Thanks @inknos.
